src/components/frame/TokenActivation/TokenActivationPopover.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { ReactElement, useMemo, useContext } from 'react';
+
+import Popover, { PopoverChildFn } from '~shared/Popover';
+import { TokenActivationContext } from '~shared/TokenActivationProvider';
+import { removeValueUnits } from '~utils/css';
+
+import TokenActivationContent from './TokenActivationContent';
+import { TokensTabProps } from './TokensTab';
+
+import styles from './TokenActivationPopover.css';
+
+interface Props extends TokensTabProps {
+  children: ReactElement | PopoverChildFn;
+}
+
+const { verticalOffset } = styles;
+
+const TokenActivationPopover = ({ children, ...otherProps }: Props) => {
+  const { isOpen, setIsOpen } = useContext(TokenActivationContext);
```

Comment:
You could create a hook for it, `useTokenActivationContext`, similar to how we wrap other contexts